File: junit5/src/test/java/io/micronaut/test/junit5/TestController.java
package io.micronaut.test.junit5;

import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;

@Controller("/test")
public class TestController {


    final TestService testService;

    TestController(TestService testService) {
        this.testService = testService;
    }

    @Get
    String index() {
        return testService.doStuff();
    }
}

File: test-kotest5/src/test/kotlin/io/micronaut/test/kotest/TestController.kt

package io.micronaut.test.kotest

import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get

@Controller("/test")
class TestController constructor(private val testService: TestService) {

    @Get
    fun index(): String {
        return testService.doStuff()
    }
}
